web/src/beta/features/Editor/DataSourceManager/utils.tsx

```diff
@@ -112,3 +113,23 @@ export const AddLayerWrapper = styled.div`
   margin-top: 5px;
   justify-content: flex-start;
 `;
+
+export const generateTitle = (url?: string, sourceType?: string): string => {
+  if (url && url.trim() !== "") {
+    try {
+      const urlObject = new URL(url);
+      const pathParts = urlObject.pathname.split("/");
+      const lastPart = pathParts.pop() || "";
+      const fileName = lastPart.split(".")[0];
+      return fileName;
+    } catch (error) {
+      console.error("Invalid URL", error);
+    }
+  }
+
+  if (sourceType === "url") {
+    console.warn("SourceType is 'url' but either URL is invalid or not provided.");
+  }
```

Comment:
Do we really need this and the sourceType param? If the try/catch gets an error we know something is wrong. Both the catch's error and this warning are goign to go off without much difference in meaning, right?

Reply:
Fair point, I'll remove it.